### ISSUE TITLE
0.13: [Bug]: `garden test` does not fail if no test action has been found, when using wildcards

### ISSUE THREAD

**Issue by stefreak** [garden-io/garden#4224]
### Current Behavior

- `garden test foobar` fails with an error `✖ Test action "foobar" was not found.`
- `garden test '*foobar'` succeeds with a warning `⚠ No Test actions found (matching argument(s) '*foobar'). Aborting.`

### Expected behavior

Both commands fail, if no test has been found. Otherwise workflows might look like they succeeded, but then no tests ran.

### Reproducible example

```
kind: Project
name: repro
environments:
 - name: default
providers:
 - name: exec
```
(No tests declared, so `garden test` should fail with any positional arguments)

### Workaround

Unknown

### Suggested solution(s)

Convert the warning to an error

### Your environment

* **OS:** macos
* **How I'm running Kubernetes:** irrelevant

`garden version` latest 0.13 edge

**Comment by Orzelius** [comment]
@edvald [requested the current behavior](https://github.com/garden-io/garden/pull/4146#discussion_r1182616857) but I actually agree with this issue. Having the test and run commands fail would be more intuitive (imo) and also unify the behavior with build and deploy commands.

**Comment by vvagaytsev** [comment]
Fixed via #4247
